hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/FixedSizeTypesCodec.java

```diff
@@ -82,8 +82,8 @@ public static void encodeInt(byte[] buffer, int pos, ProtocolType protocolType)
         encodeInt(buffer, pos, protocolType.getId());
     }
 
-    public static void encodeInt(byte[] buffer, int pos, FieldType fieldType) {
-        encodeInt(buffer, pos, fieldType.getId());
+    public static void encodeInt(byte[] buffer, int pos, FieldKind fieldKind) {
+        encodeInt(buffer, pos, fieldKind.ordinal());
```

Comment:
I would advise not using `ordinal`, but introduce explicit ids for different field kinds. As Javadoc suggests, we should avoid using it. 
https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#ordinal--